executor/pipelined_window.go

```diff
@@ -264,7 +264,7 @@ func (e *PipelinedWindowExec) getStart(ctx sessionctx.Context) (uint64, error) {
 			var res int64
 			var err error
 			for i := range e.orderByCols {
-				res, _, err = e.start.CmpFuncs[i](ctx, e.orderByCols[i], e.start.CalcFuncs[i], e.getRow(start), e.getRow(e.curRowIdx))
+				res, _, err = e.start.CmpFuncs[i](ctx, e.start.CastFuncs[i], e.start.CalcFuncs[i], e.getRow(start), e.getRow(e.curRowIdx))
```

Comment:
From the name `PipelinedWindowExec`, I assume there is a normal `WindowExec`, will this exec meet the same issue?

Reply:
> From the name `PipelinedWindowExec`, I assume there is a normal `WindowExec`, will this exec meet the same issue?

I find `window.go` will encounter the same problem. I will fix it